pkg/k8scontext/context.go

```diff
@@ -21,32 +19,47 @@ import (
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
+
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/annotations"
+	managedv1 "github.com/Azure/application-gateway-kubernetes-ingress/pkg/apis/azureingressmanagedtarget/v1"
+	prohibitedv1 "github.com/Azure/application-gateway-kubernetes-ingress/pkg/apis/azureingressprohibitedtarget/v1"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/client/clientset/versioned"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/client/informers/externalversions"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/utils"
 )
 
 // NewContext creates a context based on a Kubernetes client instance.
-func NewContext(kubeClient kubernetes.Interface, namespaces []string, resyncPeriod time.Duration) *Context {
-	var options []informers.SharedInformerOption
+func NewContext(kubeClient kubernetes.Interface, namespaces []string, resyncPeriod time.Duration, crdClient versioned.Interface) *Context {
+	updateChannel := channels.NewRingChannel(1024)
 
+	var options []informers.SharedInformerOption
+	var crdOptions []externalversions.SharedInformerOption
 	for _, namespace := range namespaces {
 		options = append(options, informers.WithNamespace(namespace))
+		crdOptions = append(crdOptions, externalversions.WithNamespace(namespace))
 	}
-
 	informerFactory := informers.NewSharedInformerFactoryWithOptions(kubeClient, resyncPeriod, options...)
+	crdInformerFactory := externalversions.NewSharedInformerFactoryWithOptions(crdClient, resyncPeriod, crdOptions...)
 
 	informerCollection := InformerCollection{
 		Endpoints: informerFactory.Core().V1().Endpoints().Informer(),
 		Ingress:   informerFactory.Extensions().V1beta1().Ingresses().Informer(),
 		Pods:      informerFactory.Core().V1().Pods().Informer(),
 		Secret:    informerFactory.Core().V1().Secrets().Informer(),
 		Service:   informerFactory.Core().V1().Services().Informer(),
+
```

Comment:
nit: remove new line

Reply:
I think a new line makes sense here - visually separating the internal Kubernetes informers from the ones that we ourselves made.  Also it helps with `go fmt` - keeps the first section narrower.

Reply:
I didn't see new line in other places where we added these two, hence!